### PR TITLE
Add organization to API Categories

### PR DESCRIPTION
**Pull request by senthuran16**
## Purpose
$Subject

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/impl/ApiCategoriesApiServiceImpl.java [hunk 1]
```diff
@@ -43,21 +44,21 @@
...
 
     public Response apiCategoriesGet(String xWSO2Tenant, MessageContext messageContext) {
-        String requestedTenantDomain = RestApiUtil.getRequestedTenantDomain(xWSO2Tenant);
+        String organization = (String) messageContext.get(RestApiConstants.ORGANIZATION);
         try {
-            if (!APIUtil.isTenantAvailable(requestedTenantDomain)) {
-                RestApiUtil.handleBadRequest("Provided tenant domain '" + xWSO2Tenant + "' is invalid",
+            if (!APIUtil.isOrganizationTenantAvailable(organization)) {
+                RestApiUtil.handleBadRequest("Tenant for the provided organization '" + organization + "' is invalid",
                         ExceptionCodes.INVALID_TENANT.getErrorCode(), log);
```
Comment (chamilaadhi) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/impl/ApiCategoriesApiServiceImpl.java:51]: if (!APIUtil.isTenantAvailable(requestedTenantDomain)) {} is not needed. this check is done from the interceptor level

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/impl/ApiCategoriesApiServiceImpl.java [hunk 2]
```diff
@@ -43,21 +44,21 @@
...
+            List<APICategory> categoryList = APIUtil.getAllAPICategoriesOfTenant(organization);
             APICategoryListDTO categoryListDTO = APICategoryMappingUtil.fromCategoryListToCategoryListDTO(categoryList);
             return Response.ok().entity(categoryListDTO).build();
         } catch (APIManagementException e) {
             String errorMessage = "Error while retrieving API categories";
             RestApiUtil.handleInternalServerError(errorMessage, e, log);
         } catch (UserStoreException e) {
-            String errorMessage = "Error while checking availability of tenant " + requestedTenantDomain;
+            String errorMessage = "Error while checking availability of tenant for organization " + organization;
             RestApiUtil.handleInternalServerError(errorMessage, e, log);
```
Comment (chamilaadhi) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/impl/ApiCategoriesApiServiceImpl.java:62]: once you remove the istenant() check, this exception will be removed

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java [hunk 3]
```diff
@@ -5058,6 +5058,16 @@ public static boolean isTenantAvailable(String tenantDomain) throws UserStoreExc
                 .isTenantActive(tenantId);
     }
 
+    public static boolean isOrganizationTenantAvailable(String organization) throws UserStoreException {
```
Comment (chamilaadhi) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java:5061]: this method won't be needed

**Review by chamilaadhi — CHANGES_REQUESTED** [review]

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/impl/ApiCategoriesApiServiceImpl.java (hunk 1 above)
Comment (senthuran16) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/impl/ApiCategoriesApiServiceImpl.java]: Done via https://github.com/wso2/carbon-apimgt/pull/10539/commits/d9368dcf8df1c26ded2c9ff81b077a5267d30044

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/impl/ApiCategoriesApiServiceImpl.java (hunk 2 above)
Comment (senthuran16) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/impl/ApiCategoriesApiServiceImpl.java]: Done via https://github.com/wso2/carbon-apimgt/pull/10539/commits/d9368dcf8df1c26ded2c9ff81b077a5267d30044

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java (hunk 3 above)
Comment (senthuran16) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java]: Done via https://github.com/wso2/carbon-apimgt/pull/10539/commits/d9368dcf8df1c26ded2c9ff81b077a5267d30044

**Review by chamilaadhi — APPROVED** [review]

**Pull request merged**